neo/rawio/nixrawio.py

```diff
@@ -242,6 +228,36 @@ def _parse_header(self):
                             event_ann.update(self._filter_properties(props, 'event'))
                             ev_idx += 1
 
+                # adding array annotations to analogsignals
+                annotated_anasigs = []
+                sig_ann = seg_ann['signals']
+                # this implementation relies on analogsignals always being
+                # stored in the same stream order across segments
+                stream_id = 0
+                for da_idx, da in enumerate(group.data_arrays):
+                    if da.type != "neo.analogsignal":
+                        continue
+                    anasig_id = da.name.split('.')[-2]
+                    # skip already annotated signals
+                    if anasig_id in annotated_anasigs:
+                        continue
+                    annotated_anasigs.append(anasig_id)
+
+                    # collect annotation properties
+                    props = [p for p in da.metadata.props
+                             if p.type != 'ARRAYANNOTATION']
+                    props_dict = self._filter_properties(props, "analogsignal")
+                    sig_ann[stream_id].update(props_dict)
+
+                    # collect array annotation properties
+                    props = [p for p in da.metadata.props
+                             if p.type == 'ARRAYANNOTATION']
+                    props_dict = self._filter_properties(props, "analogsignal")
+                    sig_ann[stream_id]['__array_annotations__'].update(
```

Comment:
I think anderstand this.

I think I don't anderstand this loop because I don't anderstand the underlying struture of nix.

`sig_ann[stream_id]['__array_annotations__'][key]` is supposed to be an array with size of channel count but here it is update insid ethe channel loop !!!

Reply:
In the current nix implementation the array_annotations are duplicated and saved for each channel individually. Therefore I only use the first channel of each analogsignal detected to add the array annotations to it.

Reply:
Once the array annotations were added to an analogsignal it won't be reannotated as it is listed in the `annotated_anasigs`

Reply:
I added a clarification in the comments. Is it now easier to understand?